src/fpsolve/type.jl

```diff
@@ -1,40 +1,34 @@
-abstract type AbstractFPSolverAlgorithm end
-abstract type AbstractFPSolverCache end
-
-@enum FPStatus::Int8 begin
-  FastConvergence     = 2
-  Convergence         = 1
-  SlowConvergence     = 0
-  VerySlowConvergence = -1
-  Divergence          = -2
-end
-
 # solver
-mutable struct FPSolver{C<:AbstractFPSolverCache,uTolType,K,C1}
+mutable struct FPSolver{algType<:DiffEqBase.AbstractNLSolverAlgorithm,IIP,uTolType,C}
+  alg::algType
   ηold::uTolType
-  κ::K
-  max_iter::Int
   fp_iters::Int
-  status::FPStatus
-  fast_convergence_cutoff::C1
+  status::DiffEqBase.NLStatus
   cache::C
 end
 
-struct FPNone end
-
-# algorithms
-struct FPFunctional{K,C} <: AbstractFPSolverAlgorithm
-  κ::K
-  fast_convergence_cutoff::C
-  max_iter::Int
+# caches
+struct FPFunctionalCache{uType,uNoUnitsType} <: DiffEqBase.AbstractNLSolverCache
+  du::uType
+  atmp::uNoUnitsType
```

Comment:
Should we also move `du` to the cache in https://github.com/JuliaDiffEq/DiffEqBase.jl/pull/315? And add a unitless cache?

Reply:
I think so. It's a bit weird to have something that is alg-specific but then is in the non-specific cache.